File: server/SERVprog/src/main/java/com/SERV/view/MapsController.java
package com.SERV.view;

import com.SERV.interfaceAbility.InterfaceMaps;
import com.SERV.interfaceAbility.UrlController;
import com.SERV.model.DataProcessing;
import com.SERV.view.entity.Event;
import com.SERV.view.entity.Maps;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.io.File;
import java.util.ArrayList;

/**
 * Created by prizrak on 25.11.2014.
 */
@Controller
@RequestMapping(UrlController.mapsObj)
public class MapsController implements InterfaceMaps{

      /*  @RequestMapping(method = RequestMethod.POST, value= "allPoint/{idMap}")
        @ResponseBody
        public List<Point> getMapData(@PathVariable int idMap) {
            return DataProcessing.getProcessingMap().getPoint(idMap);
        }*/

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapFromId_)
    @ResponseBody
    public Maps getMap(@PathVariable  int id){return DataProcessing.getProcessingMap().getMap(id);}

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapFromEventAll)
    @ResponseBody
    public ArrayList<Maps> getMaps(@RequestBody Event ev) {
        return DataProcessing.getProcessingMap().getMaps(ev);
    }

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapActivEventAll)
    @ResponseBody
    public ArrayList<Maps> getMapsFromActivEvent(){
        return DataProcessing.getProcessingMap().getMapsFromActivEvent();
    }

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapAll)
    @ResponseBody
    public ArrayList<Maps> getMaps() {
        return DataProcessing.getProcessingMap().getMaps();
    }

    @RequestMapping(method = RequestMethod.POST, consumes="application/json", produces ="application/json", value=UrlController.mapsInsert)
    @ResponseBody
    public void setMap(@RequestBody final ArrayList<Maps> map){DataProcessing.getProcessingMap().setMap(map);}

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapsDelete)
    @ResponseBody
    public void delMap(@RequestBody final ArrayList<Maps> map){DataProcessing.getProcessingMap().delMap(map);}

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapsUpdate)
    @ResponseBody
    public void updMap(@RequestBody final ArrayList<Maps> map){DataProcessing.getProcessingMap().updMap(map);}


    public void sendMapIn(byte[] file, int id){DataProcessing.getProcessingMap().sendMapIn(file, id);}

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapsSendIn)
    @ResponseBody
    public void sendMapInByte(@RequestBody final Byte[] file,@PathVariable  int id){DataProcessing.getProcessingMap().sendMapIn(byteReparser(file), id);}
   // @RequestMapping(method = RequestMethod.GET, value= UrlController.mapsSendOut)
   // @ResponseBody
    public byte[] sendMapOUT(int id){return DataProcessing.getProcessingMap().sendMapOUT(id);}

    @RequestMapping(method = RequestMethod.POST, value= UrlController.mapsSendOut)
    @ResponseBody
    public Byte[] sendMapOUTByte(@PathVariable int id){
        return byteParser(sendMapOUT(id));
    }

    private Byte[] byteParser(byte[] b){
        Byte[] bb = new Byte[b.length];
        for(int i=0; i<b.length;i++){
            bb[i]=b[i];
        }
        return bb;
    }
    private byte[] byteReparser(Byte[] b){
        byte[] bb = new byte[b.length];
        for(int i=0; i<b.length;i++){
            bb[i]=b[i];
        }
        return bb;
    }
}

File: отчёты/код/client/MapsController.cs
using Assets.myScript.entity;
using Newtonsoft.Json;
using RestSharp;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Assets.myScript.interfaceUrl
{
    class MapsController
    {

        public long mapSize(int id){
            string url = Data.getDataClass().url + InterfaceUrl.mapSize_ + id;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            client.Timeout = 5000;
            var response = client.Execute(request);
            var content = response.Content;
            try
            {
                var b = JsonConvert.DeserializeObject<long>(content);
                return b;
            }
            catch (Exception e) { return -1; }
        }

        public Maps getMap(int id)
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapFromId_+id;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            client.Timeout = 5000;
            var response = client.Execute(request);
            var content = response.Content;
            try
            {
                var b = JsonConvert.DeserializeObject<Maps>(content);
                return b;
            }
            catch (Exception e) { return null; }
        }
        public List<Maps> getMaps(Event ev) 
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapFromEventAll;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            request.AddJsonBody(ev);
            client.Timeout = 5000;
            var response = client.Execute(request);
            var content = response.Content;
            try
            {
                var b = JsonConvert.DeserializeObject<List<Maps>>(content);
                return b;
            }
            catch (Exception e) { return null; }
        }
        public List<Maps> getMaps()
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapAll;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            client.Timeout = 5000;
            var response = client.Execute(request);
            var content = response.Content;
            try
            {
                var b = JsonConvert.DeserializeObject<List<Maps>>(content);
                return b;
            }
            catch (Exception e) { return null; }
        }

        public List<Maps> getMapsFromActivEvent()
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapActivEventAll;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            client.Timeout = 5000;
            var response = client.Execute(request);
            var content = response.Content;
            try
            {
                var b = JsonConvert.DeserializeObject<List<Maps>>(content);
                return b;
            }
            catch (Exception e) { return null; }
        }

        public void setMap(List<Maps> map) 
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapsInsert;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            request.AddJsonBody(map);
            client.Timeout = 5000;
            var response = client.Execute(request);
        }
        public void sendMapIn(byte[] file, int id)
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapsSendIn_+id;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            request.AddJsonBody(file);
            client.Timeout = 5000;
            var response = client.Execute(request);
        }
        public byte[] sendMapOUT(int id) 
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapsSendOut_+id;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            client.Timeout = 7000;
            var response = client.Execute(request);
            var content = response.Content;
            try
            {
                Int32[] bi = JsonConvert.DeserializeObject<Int32[]>(content);
                byte[] b = new byte[bi.Length];
                for (int i = 0; i < bi.Length; i++)
                {
                    b[i] = (byte)bi[i];
                }
                    return b;
            }
            catch (Exception e) { return null; }
        }
        public void delMap(List<Maps> map) 
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapsDelete;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            request.AddJsonBody(map);
            client.Timeout = 5000;
            var response = client.Execute(request);
        }

        public void updMap(List<Maps> map)
        {
            string url = Data.getDataClass().url + InterfaceUrl.mapsUpdate;
            var client = new RestClient(url);
            var request = new RestRequest(Method.POST);
            request.AddJsonBody(map);
            client.Timeout = 5000;
            var response = client.Execute(request);
        }
    }
}
